src/rgw/rgw_lua_utils.h

```diff
@@ -292,10 +292,9 @@ template<typename MapType>
 typename MapType::iterator* create_iterator_metadata(lua_State* L, const std::string_view name, 
     const typename MapType::iterator& start_it, const typename MapType::iterator& end_it) {
   using Iterator = typename MapType::iterator;
-  const std::string qualified_name = get_iterator_name(name);
   // create metatable for userdata
   // metatable is created before the userdata to save on allocation if the metatable already exists
-  const auto metatable_is_new = luaL_newmetatable(L, qualified_name.c_str());
+  const auto metatable_is_new = luaL_newmetatable(L, get_iterator_name(name).c_str());
```

Comment:
@Svelar could you please reference the related Clang issue in the commit message? as the existing code is innocent.

Reply:
and, probably we should use a new Clang (>= 17.0) in our CI ?

Reply:
> @Svelar could you please reference the related Clang issue in the commit message? as the existing code is innocent.

Done, thanks for advice.

> and, probably we should use a new Clang (>= 17.0) in our CI ?

All CI machines should be Jammy I think, so clang 14 is default. Do you mean upgrade to version 17 only on CI machines (manually maybe) or like #55886 way?

Reply:
yup, only for running the CI with clang-18 by installing clang from it's own apt repo. but that'd need more discussions before starting implement it. i think.

Reply:
Yea, @dmick what do you think?

Reply:
to be more specific, i think we need to install it with jenkins job script not using `ceph.spec.in` or `debian/*` .